hipcheck/src/policy_exprs/token.rs

```diff
@@ -68,6 +81,42 @@ fn lex_float(input: &mut Lexer<'_, Token>) -> Result<F64> {
 	Ok(F64::new(f)?)
 }
 
+/// Lex a single datetime value.
+fn lex_datetime(input: &mut Lexer<'_, Token>) -> Result<Zoned> {
+	let utc = TimeZone::fixed(tz::offset(0));
+
+	let s = input.slice();
+	// Parse to a Zoned datetime value with as much detail as given
+	match s.parse::<Timestamp>() {
+		// If a UTC offset is provided, convert the datetime to the equivalent UTC datetime
+		Ok(timestamp) => Ok(timestamp.to_zoned(utc)),
+		Err(_) => match s.parse::<DateTime>() {
+			// If no offset is provided, assume the time is UTC
+			Ok(dt) => dt
+				.to_zoned(utc)
+				.map_err(|err| LexingError::InvalidDatetime(s.to_string(), JiffError::new(err))),
+			Err(_) => match s.parse::<Date>() {
+				// If no time is provided, treat the time as midnight UTC on the given day
+				Ok(date) => date.to_zoned(utc).map_err(|err| {
+					LexingError::InvalidDatetime(s.to_string(), JiffError::new(err))
+				}),
+				// If the string provided does not parse to a valid date or datetime, return an error
+				Err(err) => Err(LexingError::InvalidDatetime(
+					s.to_string(),
+					JiffError::new(err),
+				)),
+			},
+		},
+	}
```

Comment:
I think this is cleaner as a set of `if let Ok(_) = ` blocks (at the same identation level), instead of nesting the `matches`

Reply:
Would it make sense to leave the last `match` block as a `match`, since we catch the error in that instance?

Reply:
sounds good to me